sdk/src/types/block/output/mod.rs

```diff
@@ -131,6 +131,18 @@ impl OutputWithMetadata {
     }
 }
 
+/// Contains the generic [`Output`] and the associated [`OutputMetadata`].
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct OutputAndMetadata {
```

Comment:
Having both `OutputWithMetadata` and `OutputAndMetadata` is a little bit confusing

Reply:
Indeed

Reply:
How about `OutputWithProofAndMetadata`? Because otherwise the only difference is `Response` which doesn't really say anything about other fields inside

Reply:
We could do it like that, but I don't think that scales well. Honestly, right now - if `OutputAndMetadata` is not liked - I would go with `OutputWithMetadataResponse` (with the proof as defined in TIP-48) and `OutputWithMetadata` {output, metadata} without the proof.

Reply:
We could also have two structs with the same name, and differentiate based on usage with the module.

Reply:
Or even rename one locally to `OutputWithMetadataResponse` if there's a conflict.

Reply:
Let me summarize the options we have:
(1) we call { output, metadata } `OutputWithMetadata` like we used to and { output, output_id_proof, metadata } `OutputWithMetadataResponse`
(2) we call { output, metadata } `OutputWithMetadata` and { output, output_id_proof, metadata } `OutputWithMetadata` as well and differentiate them by their module path, and rename them if there's conflict.
(3) we call { output, metadata } `OutputAndMetadata` and { output, output_id_proof, metadata } `OutputWithMetadata`; additionally we re-export `OutputWithMetadata` as `OutputWithMetadataResponse` in  the `types/api/core`

==========
My take on the options:
(1) I wouldn't go for that because `OutputWithMetadataResponse` would live in `types/block/output`, so it is also an essential type, not just an API response type.
(2) possible, but I personally would find it more confusing than the proposed solution (3)
(3) this is my preferred option and the solution implemented currently in this PR

Reply:
> so it is also an essential type used in wallet/operations/syncing, not just an API response type.

What's the issue with this? Many API response types are essential for the wallet to work

Reply:
> not just an API response type

Nothing is "just" an API response type tbh, they exist to be used by the client and wallet

Reply:
hmm, since you both seem to favor (1) I'll do that then :disappointed: